src/System.Configuration.ConfigurationManager/src/System/Configuration/GenericEnumConverter.cs

```diff
@@ -26,41 +27,39 @@ public override object ConvertTo(ITypeDescriptorContext ctx, CultureInfo ci, obj
 
         public override object ConvertFrom(ITypeDescriptorContext ctx, CultureInfo ci, object data)
         {
-            object result;
-
             // For any error, throw the ArgumentException with SR.Invalid_enum_value
-            try
+            if ((data is string value) && (value.Length > 0))
             {
-                string value = (string)data;
-                if (string.IsNullOrEmpty(value)) throw new Exception();
-
-                // Disallow numeric values for enums.
-                if (!string.IsNullOrEmpty(value) &&
-                    (char.IsDigit(value[0]) ||
-                    (value[0] == '-') ||
-                    (value[0] == '+')))
-                    throw new Exception();
-
-                if (value != value.Trim())
+                // Disallow numeric values and whitespace at start and end.
+                if ((!char.IsDigit(value[0])) && (value[0] != '-') && (value[0] != '+') &&
+                    (!char.IsWhiteSpace(value[0])) && (!char.IsWhiteSpace(value[value.Length - 1])))
                 {
-                    // throw if the value has whitespace 
-                    throw new Exception();
+                    try
+                    {
+                        return Enum.Parse(_enumType, value);
+                    }
+                    catch
+                    {
+                        // Exception from parse. Will throw more appropriate exception below.
+                    }
                 }
-
-                result = Enum.Parse(_enumType, value);
             }
-            catch
-            {
-                StringBuilder names = new StringBuilder();
+            throw CreateExceptionForInvalidValue();
+        }
+
+        private ArgumentException CreateExceptionForInvalidValue()
+        {
+            StringBuilder names = new StringBuilder();
 
-                foreach (string name in Enum.GetNames(_enumType))
+            foreach (string name in Enum.GetNames(_enumType))
+            {
+                if (names.Length != 0)
                 {
-                    if (names.Length != 0) names.Append(", ");
-                    names.Append(name);
+                    names.Append(", ");
```

Comment:
Is this putting a comma at the front of the string? I see you didn't change that, but seems wrong.

Reply:
Not at the very beginning. It's the same logic as before for commas as separators, but with braces on the `if`

Reply:
Oh yes